### PR TITLE
Add recipe for hide-secrets.

### PR DESCRIPTION
**Pull request by SebastianMeisel**
### Brief summary of what the package does

Hide secrets like username, passwords, IP-addresses, public-ssh-keys etc.

### Direct link to the package repository

https://gitlab.com/ostseepinguin1/hide-secrets-el

### Your association with the package

I'm the mantainer.

### Relevant communications with the upstream package maintainer

None needed 

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

### CODE REVIEW

**Comment by progfolio** [comment]
Do you, by chance, have this repository configured to only allow clones via SSH?
I can't clone via https.

**Comment by SebastianMeisel** [comment]
That should already be possible, but as far as I know only of you have an gut lab account. If that's not working, I think I could add you as a developer to the project. You would then need to send me a public key.

File: recipes/hide-secrets [hunk 1]
```diff
@@ -0,0 +1 @@
+(hide-secrets :fetcher gitlab :repo "ostseepinguin1/hide-secets-el")
```
Comment (progfolio) [inline recipes/hide-secrets:1]: s /secets/secrets in the `:repo` string

**Comment by progfolio** [comment]
> That should already be possible, but as far as I know only of you have an gut lab account. If that's not working, I think I could add you as a developer to the project. You would then need to send me a public key.

You should see if you can configure it so it can be downloaded over https without a gitlab account. Otherwise package managers like straight.el, el-get, and Elpaca will not be able to install this package easily for end users.

**Comment by SebastianMeisel** [comment]
I'm sorry. I was not aware, that https access was restricted. I'm relativly new to gitlab (used github before). Now I works.

**Comment by riscy** [comment]
Thanks both - I modified the file in the PR to have the corrected URL as @progfolio noted.  Here's what I'm finding --

You definitely don't want to call `(hide-secrets)` inside the definition of `hide-secrets-mode`.  It's unnecessary.

`hide-secrets.el` with byte-compile using Emacs 29.3:
```
hide-secrets.el:259:40: Warning: the function `eat-mode' is not known to be defined.
```
You don't want to sharp-quote `eat-mode` in this case.

`hide-secrets.el` with checkdoc 0.6.2 (fix *within reason*):
```
hide-secrets.el:122: White space found at end of line
```
fyi :)

`hide-secrets.el` with package-lint 20240507.1722:
```
1 issue found:
39:0: error: "password-colon-equivalents" doesn't start with package's prefix "hide-secrets".
```
I think it would be better to check whether this const is bound (`boundp`) and if not, to fall back on your own default.  If every package implemented these sorts of things independently then conflicts could occur.  Alternatively you may want to look into something like the compat package and see if it has handling for this.

`hide-secrets.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- hide-secrets.el#L258: It's safer to sharp-quote function names; use `#'` e.g. `(add-hook 'post-command-hook #'hide-secrets nil t)`
- hide-secrets.el#L260: It's safer to sharp-quote function names; use `#'`
- hide-secrets.el#L262: It's safer to sharp-quote function names; use `#'`
- hide-secrets.el#L264: It's safer to sharp-quote function names; use `#'`
- hide-secrets.el#L265: It's safer to sharp-quote function names; use `#'`
- hide-secrets.el#L266: It's safer to sharp-quote function names; use `#'`
```

**Comment by SebastianMeisel** [comment]
I think, I could correct all these problems.

**Comment by riscy** [comment]
I had a look and this look good, thank you!

**Pull request merged**